test/suite/hook_tiered.py

```diff
@@ -89,29 +113,31 @@ def session_drop_replace(ntiers, orig_session_drop, session_self, uri, config):
         ret = orig_session_drop(session_self, uri, config)
     return ret
 
-# Called to replace Session.verify
-def session_verify_replace(ntiers, orig_session_verify, session_self, uri):
-    return 0
+# Called to replace Session.checkpoint.
+# We add a call to flush_tier after the checkpoint to make sure we are exercising tiered
+# functionality.  This means that tests that don't call checkpoint won't have any tiered objects.
```

Comment:
That last sentence is a bit unfortunate. What if we replaced `connection.close` to add in a `flush_tier` immediately before close?

Reply:
Nice idea.  I should be able to add that.  I'd also like to add something so we can flush part way through a test. So the test would wind up using flushed data.  Of course with this we would achieve that on tests that reopen the database.

Reply:
The obvious hook causes lots of things to break.  I _think_ this may be a bug from calling `flush_tier` with cursors open (i.e., the question of having `flush_tier` use exclusive dhandles that we discussed earlier today.). I've created WT-7683 to sort this out.